commands/host.vegman

```diff
@@ -193,35 +234,75 @@ function cmd_virtualmedia_mount {
     return 1
   fi
 
-  if [[ ! -e ${imagefile} ]] || [[ -d ${imagefile} ]]; then
-    echo "Please specify the path to image file" >&2
+  if ! [[ ${imagefile} =~ ${imagefile_pattern} ]]; then
+    echo "Please specify the valid path to the image file" >&2
     return 1
   fi
 
-  usb-ctrl insert "usb.ms.cli" "${imagefile}" "${type}"
+  if [[ ! -z ${username} ]] || [[ ! -z ${password} ]]
+  then
+    local cred_pipe_fifo=$(mktemp -u)
+    cred_pipe=6
+    mkfifo $cred_pipe_fifo
+    exec 6<>$cred_pipe_fifo
+    rm $cred_pipe_fifo
+
+    printf %b "${username:0:64}\0${password:0:64}\0\n" >&6
+  fi
+
+  $(busctl call xyz.openbmc_project.VirtualMedia \
+    /xyz/openbmc_project/VirtualMedia/Legacy/USB1 \
+    xyz.openbmc_project.VirtualMedia.Legacy \
+    Mount sbyv "$imagefile" $readonly ${type} y $cred_pipe  >/dev/null) && \
```

Comment:
But I still don't understand how can it work?
`virtual-media` is not our child process and it knows nothing about our local file descriptors.
Probably, we should specify here the full path to the named pipe.

Reply:
OK, I found out that it is an internal `systemd` possibilities. There is a special D-Bus type `unix_fd` that can be passed between client and service. And `libsystemd` likely handles it correct.

Let's assume that it works this way.